### PR TITLE
change Proxy pattern

### PR DESCRIPTION
**Pull request by ko22009**
Current realization isn't correctly.

### CODE REVIEW

File: Structural/Proxy/RecordInterface.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+{
+    /**
+     * @param string $name
+     * @param string $value
+     */
+    public function __set(string $name, string $value);
+
+    /**
+     * @param string $name
+     * @return string
```
Comment (Goussema) [inline Structural/Proxy/RecordInterface.php]: line break here before return

File: Structural/Proxy/RecordProxy.php [hunk 2]
```diff
@@ -2,48 +2,36 @@
...
 
     /**
      * @param string $name
-     * @param string  $value
+     * @param string $value
      */
     public function __set(string $name, string $value)
     {
-        $this->isDirty = true;
+        if ($this->record == null) {
```
Comment (Goussema) [inline Structural/Proxy/RecordProxy.php]: what about if (null === $this->record )

File: Structural/Proxy/RecordProxy.php [hunk 3]
```diff
@@ -2,48 +2,36 @@
...
 
-        parent::__set($name, $value);
+        $this->record->__set($name, $value);
     }
 
-    public function isDirty(): bool
+    public function __get(string $name): string
     {
-        return $this->isDirty;
+        if ($this->record == null) {
```
Comment (Goussema) [inline Structural/Proxy/RecordProxy.php]: same here

**Comment by domnikl** [comment]
As I refactored the Proxy pattern in ea85485b85a4fba39ca788f6bec74342f2a78087 without the use of magic methods and with a nicer and more real-world-like example, I am gonna close this.

**Pull request closed without merging**